### ISSUE TITLE
Non-interrupting start events missing in create anything list

### ISSUE THREAD

**Issue by nikku** [bpmn-io/bpmn-js-create-append-anything#4]
### Describe the Bug

As a user I want to create a non interrupting message start event in an event sub-process, but I cannot find it.

![image](https://user-images.githubusercontent.com/58601/218051649-434d7540-2f04-4f91-ae6b-2667c081f7cc.png)

### Steps to Reproduce

1. Trigger create menu
2. Search for `non`
3. Non interrupting start events do not appear in the list

### Expected Behavior

Non interrupting start events appear once searched for. I don't think it is strictly necessary to show them all the time.

### Environment

 - Browser: Any
 - OS: Any
 - Library version: 11.3.0

**Comment by nikku** [comment]
CC @smbea

**Comment by barmac** [comment]
There is a hard-coded check which removes Participants and Start Events from append menu.

**Comment by nikku** [comment]
Instead of doing that we could hide them initially (`rank=-1`). We also allow "appending" of event sub-processes.

However this issue specifically is about the create menu, where these entries are not available, too.

**Comment by smbea** [comment]
Yes the non interrupting start event is missing completely from the options list

**Comment by smbea** [comment]
I started looking at this. All the non-interrupting start events are missing.

With the current rules, these events are changed into default start events when not placed in an event sub process:

https://user-images.githubusercontent.com/25825387/219695027-54beb9b3-8c46-4ee1-aa9a-59a6e2f76ccd.mov


In my opinion, I think it makes sense to instead block the shape placement like we do for the non-interrupting boundary events. Demo:

https://user-images.githubusercontent.com/25825387/219694934-9b045ed0-7ba5-480a-baab-ec0995df5a13.mov

Wdyt?

**Comment by barmac** [comment]
Instead of blocking interaction, I'd suggest to implement a proper create preview, cf. https://github.com/bpmn-io/bpmn-js/pull/2329#issuecomment-2979960294

Also, I'd rather expect a non-interrupting message event to be converted to an interrupting one, but that could be a stretch goal.

**Comment by Buckwich** [comment]
> Also, I'd rather expect a non-interrupting message event to be converted to an interrupting one, but that could be a stretch goal.

there's an issue with some other rules rejecting replacing a non interrupting start event (https://github.com/bpmn-io/bpmn-js/issues/2313) I've also looked into, I can take this into account aswell

**Comment by nikku** [comment]
> Also, I'd rather expect a non-interrupting message event to be converted to an interrupting one, but that could be a stretch goal.

Let's consider the UX, test it and see what works well / what is surprising.
